scripts/beekeeper.pl

```diff
@@ -507,6 +514,47 @@ sub register_beekeeper {
     return $beekeeper;
 }
 
+
+sub big_red_button {
+    my ( $self, $valley ) = @_;
+
+    my $bk_a = $self->{dba}->get_BeekeeperAdaptor();
+    my $blocked_beekeepers;
+
+    # Save a list of IDs of beekeepers which were blocked earlier so
+    # that we can not mention them while reporting the current blocking.
+    $blocked_beekeepers = $bk_a->fetch_all( 'is_blocked = 1' );
+    my %previously_blocked_ids;
+    while ( my $blocked_bk = shift @{ $blocked_beekeepers } ) {
+        $previously_blocked_ids{ $blocked_bk->dbID() } = 1;
+    }
+
+    # Begin the shutdown by blocking all registered beekeepers so that
+    # none of them start spawning new workers just as this one tries to
+    # kill all workers.
+    $bk_a->block_all_alive_beekeepers();
+
+    # Report which beekeepers we have just blocked
+    $blocked_beekeepers = $bk_a->fetch_all( 'is_blocked = 1' );
+    my @currently_blocked = grep {
+        ! exists $previously_blocked_ids{ $_->dbID() }
+    } @{ $blocked_beekeepers };
+    while ( my $blocked_bk = shift @currently_blocked ) {
+        print 'Blocked beekeeper ' . $blocked_bk->dbID() . ': '
```

Comment:
Maybe just exclude the red-button bekeeper there ?

Reply:
Good idea.